builder/azure/arm/config.go

```diff
@@ -303,6 +303,8 @@ func setCloudEnvironment(c *Config) error {
 	switch name {
 	case "CHINA", "CHINACLOUD", "AZURECHINACLOUD":
 		c.cloudEnvironment = &azure.ChinaCloud
+	case "GERMAN", "GERMANCLOUD", "AZUREGERMANCLOUD":
```

Comment:
How do you feel about adding this switch statement as a helper function into autorest directly?

(Otherwise I'm going to copy-paste it for my own uses...)